drivers/frequency/adf4371/adf4371.c

```diff
@@ -0,0 +1,820 @@
+	{ ADF4371_REG(0x39), 0x07 },
+	{ ADF4371_REG(0x3A), 0x55 },
+	{ ADF4371_REG(0x3E), 0x0C },
+	{ ADF4371_REG(0x3F), 0x80 },
+	{ ADF4371_REG(0x40), 0x50 },
+	{ ADF4371_REG(0x41), 0x28 },
+	{ ADF4371_REG(0x47), 0xC0 },
+	{ ADF4371_REG(0x52), 0xF4 },
+	{ ADF4371_REG(0x70), 0x03 },
+	{ ADF4371_REG(0x71), 0x60 },
+	{ ADF4371_REG(0x72), 0x32 },
+};
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/**
+ * SPI register write to device.
+ * @param dev - The device structure.
+ * @param reg - The register address.
+ * @param val - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_write(struct adf4371_dev *dev,
+			     uint16_t reg,
+			     uint8_t val)
+{
+	uint8_t buf[3];
+	uint16_t cmd;
+
+	cmd = ADF4371_WRITE | ADF4371_ADDR(reg);
+	buf[0] = cmd >> 8;
+	buf[1] = cmd & 0xFF;
+	buf[2] = val;
+
+	return spi_write_and_read(dev->spi_desc, buf, ARRAY_SIZE(buf));
+}
+
+/**
+ * SPI register write bulk to device.
+ * @param dev - The device structure.
+ * @param reg - The register address.
+ * @param val - The buffer of data to be written.
+ * @param size - The buffer size.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_write_bulk(struct adf4371_dev *dev,
+				  uint16_t reg,
+				  uint8_t *val,
+				  uint8_t size)
+{
+	uint8_t buf[10];
+	uint16_t cmd;
+	uint8_t i;
+
+	cmd = ADF4371_WRITE | ADF4371_ADDR(reg);
+	buf[0] = cmd >> 8;
+	buf[1] = cmd & 0xFF;
+	for (i = 0; i < size; i++)
+		buf[2 + i] = val[i];
+
+	return spi_write_and_read(dev->spi_desc, buf, 2 + size);
+}
+
+/**
+ * SPI register read from device.
+ * @param dev - The device structure.
+ * @param reg - The register address.
+ * @param val - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_read(struct adf4371_dev *dev,
+			    uint16_t reg,
+			    uint8_t *val)
+{
+	uint8_t buf[3];
+	uint16_t cmd;
+	int ret;
+
+	cmd = ADF4371_READ | ADF4371_ADDR(reg);
+	buf[0] = cmd >> 8;
+	buf[1] = cmd & 0xFF;
+	buf[2] = 0;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, ARRAY_SIZE(buf));
+	if (ret < 0)
+		return ret;
+
+	*val = buf[2];
+
+	return SUCCESS;
+}
+
+/**
+ * SPI register update.
+ * @param dev - The device structure.
+ * @param reg - The register address.
+ * @param reg - The register address.
+ * @param val - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+static int32_t adf4371_update(struct adf4371_dev *dev,
+			      uint16_t reg,
+			      uint8_t mask,
+			      uint8_t val)
+{
+	uint8_t read_val;
+	int32_t ret = 0;
+
+	ret = adf4371_read(dev, reg, &read_val);
+	read_val &= ~mask;
+	read_val |= val;
+	ret |= adf4371_write(dev, reg, read_val);
```

Comment:
It would be better for the principle of failing fast to check the status of the function immediately after return. My suggestion for the code would be:
```suggestion
	ret = adf4371_read(dev, reg, &read_val);
	if (IS_ERR_VALUE(ret))
		return FAILURE; //or return ret;
	read_val &= ~mask;
	read_val |= val;
	return adf4371_write(dev, reg, read_val);
```